mmf_cli/run.py

```diff
@@ -30,9 +30,10 @@ def main(configuration, init_distributed=False, predict=False):
     registry.register("seed", config.training.seed)
     print(f"Using seed {config.training.seed}")
 
-    registry.register("writer", Logger(config, name="mmf.train"))
-
     trainer = build_trainer(configuration)
+
+    # Logger should be registered after config is registered in build_trainer
+    registry.register("writer", Logger(config, name="mmf.train"))
```

Comment:
This is wrong. This will override the logger registered by the `build_trainer` function. We did it specifically so that if something needs to log before trainer is initialized, they are able to use it using this this custom logger with name `mmf.train`. This will be overridden by build_trainer function and will be replaced. So, this needs to go before `build_trainer`.

Reply:
`build_trainer` doesn't register a logger. trainer.load() does. So this change doesn't affect that behavior. Before we initialize logger we need `config` to be registered, which is done in `build_trainer`. Otherwise when Logger initialization calls `get_env_key` it will get default config values. So currently logs are always saved in `./save/logs` irrespective of whether the `save_dir` is overidden or not. So either we can do it this way or move config registration out of `build_trainer` class. Third way is to use save_dir, log_dir directly from config passed to logger.